com.microsoft.mrtk.windowsspeech/Subsystems/TextToSpeech/WindowsTextToSpeechSubsystem.cs

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Subsystems;
+using System;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using UnityEngine;
+using UnityEngine.Scripting;
+#if WINDOWS_UWP
+using System.Threading.Tasks;
+using Windows.Foundation;
+using Windows.Media.SpeechSynthesis;
+using Windows.Storage.Streams;
+#endif // WINDOWS_UWP
+
+namespace Microsoft.MixedReality.Toolkit.Speech.Windows
+{
+    [Preserve]
+    [MRTKSubsystem(
+        Name = "com.microsoft.mrtk.windowsspeech.texttospeech",
+        DisplayName = "Windows Text-To-Speech Subsystem",
+        Author = "Microsoft",
+        ProviderType = typeof(WindowsTextToSpeechSubsystemProvider),
+        SubsystemTypeOverride = typeof(WindowsTextToSpeechSubsystem),
+        ConfigType = typeof(BaseSubsystemConfig))]
+    public class WindowsTextToSpeechSubsystem : TextToSpeechSubsystem
+    {
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+        static void Register()
+        {
+            // Fetch subsystem metadata from the attribute.
+            var cinfo = XRSubsystemHelpers.ConstructCinfo<WindowsTextToSpeechSubsystem, TextToSpeechSubsystemCinfo>();
+
+            if (!WindowsTextToSpeechSubsystem.Register(cinfo))
+            {
+                Debug.LogError($"Failed to register the {cinfo.Name} subsystem.");
+            }
+        }
+
+        [Preserve]
+        class WindowsTextToSpeechSubsystemProvider : Provider
+        {
+            public WindowsTextToSpeechSubsystemProvider() : base()
+            { }
+
+            public override void Destroy()
+            {
+#if WINDOWS_UWP
+                if (synthesizer != null)
+                {
+                    synthesizer.Dispose();
+                    synthesizer = null;
+                }
+#endif // WINDOWS_UWP
+            }
+
+            #region ITextToSpeechSubsystem implementation
+
+#if !(WINDOWS_UWP || UNITY_STANDALONE_WIN || UNITY_EDITOR_WIN)
+            private bool haveLogged = false;
+#endif
+
+            /// <inheritdoc/>
+            public override async Task<bool> TrySpeak(string phrase, AudioSource audioSource)
+            {
+                if (audioSource == null)
+                {
+                    Debug.LogError("Must specify the AudioSource object on which the text to speech data should be applied.");
+                    return false;
+                }
+
+                byte[] waveData = await Synthesize(phrase);
```

Comment:
Since all of these array allocations stay (mostly) internal to this implementation (both the synthesis array and the conversion array below), I've been trying to come up with ideas for reducing the allocations...the main one I'm thinking about at the moment is https://learn.microsoft.com/en-us/dotnet/api/system.buffers.arraypool-1, but I think we'd need to adopt 2021 as our min version to use that.

Mostly a thought experiment for now though, since this is basically all internal and we can adjust going forward.